pkg/servicerepo/servicerepo.go

```diff
@@ -0,0 +1,58 @@
+package servicerepo
+
+/**
+ * ServiceRepository: a simple repository of string-keyed interfaces,
+ * intended to be used as repo for internal services/components, to be used by each other.
+ * Intended usage:
+ * - Create one instance of the repo, early on.
+ * - Pass this repo to service initialization, in a tree of calls.  This may seem cumbersome, but it ensures determined initialization order (avoid circular dependencies)
+ * - Services should add themself to the repo, by interface.  Service type should be private, only interface methods should be public.
+ * - Services can take out other services they need, they should store them in local variables.
+ * - If interfaces are added, there is no dependency to implementations, mocking is easy to achieve.
+ */
+
+import (
+	"reflect"
+)
+
+type ServiceRepo struct {
+	s map[string]interface{}
+}
+
+var singleInstance *ServiceRepo
```

Comment:
servicerepo is a strange package name, also cannot understand why do we need our own implementation, 
I suggest to look at https://github.com/avelino/awesome-go#dependency-injection

Reply:
Good point.  I dropped own, and switched to gocontainer (from list above).